rotkehlchen/chain/evm/decoding/curve/decoder.py

```diff
@@ -352,39 +417,53 @@ def _decode_curve_trades(self, context: DecoderContext) -> DecodingOutput:
         sold_token_address: ChecksumEvmAddress | None = None
         bought_token_address: ChecksumEvmAddress | None = None
 
-        swapping_contract: ChecksumEvmAddress
-        if context.tx_log.topics[0] in (TOKEN_EXCHANGE, TOKEN_EXCHANGE_UNDERLYING):
-            pool_address = context.tx_log.address
-            swapping_contract = pool_address
-            # When a single pool is used, spender and receiver is always the same
-            spender_address = receiver_address = hex_or_bytes_to_address(context.tx_log.topics[1])
-            sold_token_id = hex_or_bytes_to_int(context.tx_log.data[:32])
-            raw_sold_amount = hex_or_bytes_to_int(context.tx_log.data[32:64])
-            bought_token_id = hex_or_bytes_to_int(context.tx_log.data[64:96])
-            raw_bought_amount = hex_or_bytes_to_int(context.tx_log.data[96:128])
-            if (
-                context.tx_log.topics[0] == TOKEN_EXCHANGE and
-                pool_address in self.pools and
-                len(self.pools[pool_address]) > max(sold_token_id, bought_token_id)  # Make sure that tokens of the pool are cached  # noqa: E501
-            ):
-                sold_token_address = self.pools[pool_address][sold_token_id]
-                bought_token_address = self.pools[pool_address][bought_token_id]
-        else:  # EXCHANGE_MULTIPLE
-            swapping_contract = self.curve_swap_router
+        swapping_contracts: set[ChecksumEvmAddress] = set()
+        # we check if before using the router any pool is used separately.
```

Comment:
IS this comment still valid? Why are you iterating all logs here? This makes no sense.

This function is called for 

```python
        if context.tx_log.topics[0] in (
            TOKEN_EXCHANGE,
            TOKEN_EXCHANGE_UNDERLYING,
            TOKEN_EXCHANGE_NG,
            EXCHANGE_MULTIPLE,
            EXCHANGE_NG,
        ):
```

And then in here you reiterate all logs and then have a similar (subset) if condition?



Also does the function's docstring still make sense? It seems to describe something different? Can you make sure this code make sense?